src/Evolution/Systems/Cce/AnalyticSolutions/RobinsonTrautman.cpp

```diff
@@ -0,0 +1,404 @@
+    make_const_view(make_not_null(&rt_scalar_reference.data()), local_rt_scalar,
+                    0, local_rt_scalar.size());
+    SpinWeighted<ComplexDataVector, 0> du_rt_scalar_reference;
+    du_rt_scalar_reference.set_data_ref(local_du_rt_scalar.data(),
+                                        local_du_rt_scalar.size());
+    du_rt_scalar(make_not_null(&du_rt_scalar_reference), rt_scalar_reference);
+    Spectral::Swsh::filter_swsh_boundary_quantity(
+        make_not_null(&du_rt_scalar_reference), l_max_, l_max_ - 3);
+  };
+  step_range_ = stepper_.do_step(rt_system);
+}
+
+std::unique_ptr<WorldtubeData> RobinsonTrautman::get_clone() const noexcept {
+  return std::make_unique<RobinsonTrautman>(*this);
+}
+
+void RobinsonTrautman::prepare_solution(const size_t l_max,
+                                        const double time) const noexcept {
+  ASSERT(l_max == l_max_,
+         "The Robinson-Trautman solution only supports the l_max resolution "
+         "specified at construction, as it must internally store the evolved "
+         "scalar at the desired resolution");
+  if (step_range_.first > time) {
+    ERROR(
+        "The Robinson-Trautman solution does not support stepping backwards in "
+        "time during the internal evolution. This error likely means that the "
+        "corresponding evolution test should be using a multistep time-stepper "
+        "so that the requested boundary data is monotonic.");
+  }
+  // step until the target time is within the current timestep
+  const auto rt_system = [this](const ComplexDataVector& local_rt_scalar,
+                                ComplexDataVector& local_du_rt_scalar,
+                                const double /*t*/) noexcept {
+    local_du_rt_scalar.destructive_resize(local_rt_scalar.size());
+    const SpinWeighted<ComplexDataVector, 0> rt_scalar_reference;
+    make_const_view(make_not_null(&rt_scalar_reference.data()), local_rt_scalar,
+                    0, local_rt_scalar.size());
+    SpinWeighted<ComplexDataVector, 0> du_rt_scalar_reference;
+    du_rt_scalar_reference.set_data_ref(local_du_rt_scalar.data(),
+                                        local_du_rt_scalar.size());
+    du_rt_scalar(make_not_null(&du_rt_scalar_reference), rt_scalar_reference);
+    Spectral::Swsh::filter_swsh_boundary_quantity(
+        make_not_null(&du_rt_scalar_reference), l_max_, l_max_ - 3);
+  };
+  while (step_range_.second < time) {
+    step_range_ = stepper_.do_step(rt_system);
+  }
+  stepper_.calc_state(time, get(dense_output_rt_scalar_).data());
+  du_rt_scalar(make_not_null(&get(dense_output_du_rt_scalar_)),
+               get(dense_output_rt_scalar_));
+  Spectral::Swsh::filter_swsh_boundary_quantity(
+      make_not_null(&get(dense_output_du_rt_scalar_)), l_max_, l_max_ - 3);
+}
+
+void RobinsonTrautman::variables_impl(
+    const gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, -2>>*> news,
+    const size_t l_max, const double /*time*/,
```

Comment:
Is it safe that the `time` matches the internal state of the class? Perhaps check that?